stable/database/values.yaml

```diff
@@ -323,7 +340,7 @@ database:
       ## settings for the journal backup
       journalBackup:
         enabled: false
-        intervalMinutes: 15
+        journalSchedule: "?/15 * * * *"
```

Comment:
> Could you please explain the benefit of changing the journalBackup.intervalMinutes to a cron schedule?

This is done so that all backup schedule options are consistent and it's easy for us and for the users to spot them.
We already have _cron schedules_ exposed for other backup types so it is not something new that the user should take into consideration and learn how to define.  It also aligns better when used with named backup groups. I.e 

```
backupGroups:
  foo:
    labels: "..."
    fullSchedule: "00 03 * * 6"
    incrementalSchedule: "00 03 * * 0-5"
    journalSchedule: "?/5 * * * *"
```

I see that Brian also likes the change but if someone else thinks that this is not appropriate, I am fine with having a further discussion.

> I am a firm believer in giving users controls which are reasonably self-explanatory, and which prevent - as much as is reasonable - the users from setting an insane value.

According to me, the new option name of `journalSchedule` is as descriptive as the old one. I don't think that the option name of `intervalMinutes` prevents the user to configure "insane" values (you can still put 120 or more). Maybe 99% of the time there won't be a need to change the default value. If we want to make it clear that journal backup should be taken often then we need to document it properly in the `values.yaml` and the `README.md`. I've updated them so that it's more clear.

Reply:
The value of a cron schedule has 5 fields - of which one, and only one, (minutes) is valid for journal backups.

If the user sets a value _in any of the other 4 fields_ the result will most likely be crashing SMs.

If you give users fields to play with which are, in fact, illegal to use - they are going to ask why they were given that ability in the first place.

>I don't think that the option name of intervalMinutes prevents the user to configure "insane" values (you can still put 120 or more).
It is very easy to put a check into the helm charts to limit the values of intervalMinutes to be between 5 and some upper value. (what should that upper value be?)
It will be much more effort to the build a check that does the same for a cron schedule value - you would need to pull the cron schedule apart, and throw an error if any field other than minutes is set _and_ check that the minutes filed is within sane limits.

> This is done so that all backup schedule options are consistent
But it masks the very real fact that the appropriate values for journal backups are _NOT_ consistent with the other backups
There is no situation in which a schedule other than some multiple of minutes is reasonable or safe for the journal backup period.

Note the following inconsistencies of journal backups vs full or incremental backups:
* if an incremental backups fail then the database will continue running with no problems;
* if a full backup fails, then the database will continue running - unless that failure impacts journal backups;
* if a journal backup fails repeatedly, then the backup SMs will run out of journal disk space and crash;
   - this may bring the entire database down, or eliminate its HA component;
 * if a full or incremental backup is configured with an infrequent period, then the result is relatively few backups with large gaps between them;
 * if a journal backup is configured with an infrequent period - the backup SMs crash.

In short, I don't believe that consistency overrules practicality.
A consistently bad implementation is still a bad implementation.

Reply:
... All that being said, I guess that the user-settable parameters of the helm charts become less of an issue as we move towards DBaaS...